pkg_resources/__init__.py

```diff
@@ -2958,8 +2959,11 @@ def _find_adapter(registry, ob):
 def ensure_directory(path):
     """Ensure that the parent directory of `path` exists"""
     dirname = os.path.dirname(path)
-    if not os.path.isdir(dirname):
+    try:
```

Comment:
This is good, and I was going to accept it straight-up, but given that this technique is superseded in Python 3.2 with the [exists_ok parameter](https://docs.python.org/3.2/library/os.html#os.makedirs). Therefore, I'd prefer to use the py27compat or py31compat module to house this legacy approach. Can you put it in py31compat (as a makedirs function that accepts exists_ok kwarg)?